image/default_image.go

```diff
@@ -125,45 +125,68 @@ func (d DefaultImageService) Login(RegistryURL, RegistryUsername, RegistryPasswd
 
 func (d DefaultImageService) Delete(imageName string) error {
 	var (
-		images []*v1.Image
-		image  *v1.Image
+		images        []*v1.Image
+		image         *v1.Image
+		imageTagCount int
 	)
 	named, err := reference.ParseToNamed(imageName)
 	if err != nil {
 		return err
 	}
 
-	image, err = imageutils.GetImage(named.Raw())
+	imageMetadataMap, err := imageutils.GetImageMetadataMap()
 	if err != nil {
 		return err
 	}
+	imageMetadata, ok := imageMetadataMap[named.Raw()]
+	if !ok {
+		return fmt.Errorf("failed to find image with name %s", imageName)
+	}
+	err = imageutils.DeleteImage(imageName)
+	if err != nil {
+		return fmt.Errorf("failed to untag image %s, err: %s", imageName, err)
+	}
+	logger.Info("untag image %s succeeded", imageName)
 
-	imageMetadataMap, err := imageutils.GetImageMetadataMap()
+	for _, value := range imageMetadataMap {
+		if value.ID == imageMetadata.ID {
+			imageTagCount++
+			if imageTagCount > 1 {
+				break
+			}
+		}
+	}
+	if imageTagCount != 1 {
+		return nil
+	}
+
+	image, err = imageutils.GetImageByID(imageMetadata.ID)
```

Comment:
what is the point for getting image? imageName and imageId are in imageMetadata already.

Reply:
It need image's detailed content which was saved in image yaml to delete a image.